src/core/Akka/IO/TcpConnection.cs

```diff
@@ -560,6 +561,12 @@ private void ReleaseSocketAsyncEventArgs()
                 Tcp.SocketEventArgsPool.Release(SendArgs);
                 SendArgs = null;
             }
+
+            if (ConnectArgs != null)
+            {
+                Tcp.SocketEventArgsPool.Release(ConnectArgs);
```

Comment:
I'm fine with this solution, however the initial intention here was to recycle connectArgs after connection has been established. After that point it won't be used anymore, so it doesn't need to be keep alive until actor stops.